File: algorithms/SORTING/bubbleSort/.cpp
//algo - sorting - bubblesort

#include<bits/stdc++.h>

using namespace std;

void bubbleSort(int a[100], int length){
	for(int i = 0; i < length - 1; i++){
		for(int j = 0; j < length - 1 - i; j++){
			if(a[j] > a[j + 1]){
				swap(a[j], a[j + 1]);
				//using third index (tmp);
			}
		}
	}
}

void show(int a[100], int length){
	for(int i = 0; i < length; i++){
		cout << a[i] << " ";
	}
	cout << endl;
}

int main(){
	int t;
	cin >> t;
	int a[100];
	for(int i = 0; i < t; i++){
		cin >> a[i];
	}
	cout << "Array before: " << endl;
	show(a, t);
	
	cout << "Array after: " << endl;
	
	bubbleSort(a, t); //sorting array;
	show(a, t); // print out array after sorted;
	return 0;
}

File: algorithms/SORTING/bubbleSort/.py
list = [x for x in range(10, 0, -1)]

def bubbleSort(list):
    length = len(list)

    for i in range(length - 1):
        for j in range(length - 1 - i):
            if list[j] > list[j + 1]:
                list[j], list[j + 1] = list[j + 1], list[j]

    return list 
list = bubbleSort(list)
print(list)